### PR TITLE
Adding API1600 support for Ethernet/FC/FCoE network resources

### PR DESCRIPTION
**Pull request by yuvirani**
### Description
Adding API1600 support for Ethernet/FC/FCoE network resources

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass (`$ rake test`).
- [x] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [x] New endpoints supported are updated in the endpoints-support.md file.
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: examples/shared_samples/ethernet_network.rb [hunk 1]
```diff
@@ -37,7 +37,7 @@
 ethernet_class = OneviewSDK.resource_named('EthernetNetwork', @client.api_version)
 
 # Scope class used in this sample
-scope_class = OneviewSDK.resource_named('Scope', @client.api_version) unless @client.api_version.to_i <= 200
+scope_class = OneviewSDK.resource_named('Scope', 1200) unless @client.api_version.to_i <= 200
```
Comment (chebroluharika) [inline examples/shared_samples/ethernet_network.rb:40]: please don't hardcode this apiversion here

File: examples/shared_samples/fc_network.rb [hunk 2]
```diff
@@ -37,14 +37,14 @@
 fc_network_class = OneviewSDK.resource_named('FCNetwork', @client.api_version)
 
 # Scope class used in this sample
-scope_class = OneviewSDK.resource_named('Scope', @client.api_version) unless @client.api_version.to_i <= 200
+scope_class = OneviewSDK.resource_named('Scope', 1200) unless @client.api_version.to_i <= 200
```
Comment (chebroluharika) [inline examples/shared_samples/fc_network.rb:40]: same as above comment

File: endpoints-support.md [hunk 3]
```diff
@@ -74,15 +74,15 @@ OneviewSDK::Datacenter.find_by(@client, width: 11000).map(&:remove)
...
+|<sub>/rest/ethernet-networks/bulk</sub>                                                  | POST        | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
+|<sub>/rest/ethernet-networks/{id}</sub>                                                  | GET         | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
+|<sub>/rest/ethernet-networks/{id}</sub>                                                  | PUT         | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
+|<sub>/rest/ethernet-networks/{id}</sub>                                                  | PATCH       | :white_check_mark:   | :white_check_mark:   | :heavy_minus_sign:   | :heavy_minus_sign:   | :heavy_minus_sign:   | :heavy_minus_sign:   | :white_check_mark:   |
```
Comment (chebroluharika) [inline endpoints-support.md:82]: I don't see this endpoint in API docs

File: endpoints-support.md [hunk 4]
```diff
@@ -93,19 +93,19 @@ OneviewSDK::Datacenter.find_by(@client, width: 11000).map(&:remove)
...
+|<sub>/rest/fc-networks</sub>                                                             | GET         | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
+|<sub>/rest/fc-networks</sub>                                                             | POST        | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
+|<sub>/rest/fc-networks/{id}</sub>                                                        | GET         | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
+|<sub>/rest/fc-networks/{id}</sub>                                                        | PATCH       | :white_check_mark:   | :white_check_mark:   | :heavy_minus_sign:   | :heavy_minus_sign:   | :heavy_minus_sign:   | :heavy_minus_sign:   | :white_check_mark:   |
```
Comment (chebroluharika) [inline endpoints-support.md:99]: I don't see this endpoint in APIdocs. Please check.

File: endpoints-support.md [hunk 5]
```diff
@@ -93,19 +93,19 @@ OneviewSDK::Datacenter.find_by(@client, width: 11000).map(&:remove)
...
+|<sub>/rest/fcoe-networks</sub>                                                           | GET         | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
+|<sub>/rest/fcoe-networks</sub>                                                           | POST        | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
+|<sub>/rest/fcoe-networks/{id}</sub>                                                      | GET         | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
+|<sub>/rest/fcoe-networks/{id}</sub>                                                      | PATCH       | :white_check_mark:   | :white_check_mark:   | :heavy_minus_sign:   | :heavy_minus_sign:   | :heavy_minus_sign:   | :heavy_minus_sign:   | :white_check_mark:   |
```
Comment (chebroluharika) [inline endpoints-support.md:106]: I don't see this endpoint in API docs. Please check.

File: examples/shared_samples/fcoe_network.rb [hunk 6]
```diff
@@ -36,7 +36,7 @@
 fcoe_network_class = OneviewSDK.resource_named('FCoENetwork', @client.api_version)
 
 # Scope class used in this sample
-scope_class = OneviewSDK.resource_named('Scope', @client.api_version) unless @client.api_version.to_i <= 200
+scope_class = OneviewSDK.resource_named('Scope', 1200) unless @client.api_version.to_i <= 200
```
Comment (chebroluharika) [inline examples/shared_samples/fcoe_network.rb:39]: Please remove this hardcoded apiversion.

File: examples/shared_samples/ethernet_network.rb (hunk 1 above)
Comment (yuvirani) [reply examples/shared_samples/ethernet_network.rb]: Reverted back

File: examples/shared_samples/fc_network.rb (hunk 2 above)
Comment (yuvirani) [reply examples/shared_samples/fc_network.rb]: Done

File: endpoints-support.md (hunk 3 above)
Comment (yuvirani) [reply endpoints-support.md]: Removed PATCH for 1600

File: endpoints-support.md (hunk 4 above)
Comment (yuvirani) [reply endpoints-support.md]: Removed

File: examples/shared_samples/fcoe_network.rb (hunk 6 above)
Comment (yuvirani) [reply examples/shared_samples/fcoe_network.rb]: Done

File: endpoints-support.md (hunk 5 above)
Comment (yuvirani) [reply endpoints-support.md]: Removed

**Pull request merged**